packages/cms/src/app/api/auth/callback/route.ts

```diff
@@ -0,0 +1,101 @@
+import { decodeIdToken } from 'arctic'
+import { cookies } from 'next/headers'
+import { google } from '~/server/auth/google'
+
+import type { OAuth2Tokens } from 'arctic'
+import { z } from 'zod'
+import { env } from '~/env'
+import { setSession } from '~/server/auth/cookie'
+import { AuthError } from '~/lib/auth-errors'
+
+const claimsSchema = z.object({
+  sub: z.string(),
+  hd: z.string().optional(),
+  email: z.string().email(),
+  email_verified: z.boolean(),
+  name: z.string(),
+  picture: z.string().optional(),
+})
+
+// 12 hours
+const expiresAt = new Date(Date.now() + 1000 * 60 * 60 * 12)
```

Comment:
Are route files evaluated each time somebody calls an endpoint? If not, then `expirestAt` will be evaluated once and will become obsolete within 12 hours after the deployment, no?

Reply:
uhm, yeah, good catch!